python/lancedb/query.py

```diff
@@ -488,6 +488,25 @@ def to_arrow(self) -> pa.Table:
         scores = pa.array(scores)
         output_tbl = self._table.to_lance().take(row_ids, columns=self._columns)
         output_tbl = output_tbl.append_column("score", scores)
+
+        if self._where is not None:
+            try:
+                # TODO would be great to have Substrait generate pyarrow compute expressions
+                # or conversely have pyarrow support SQL expressions using Substrait
+                import duckdb
+
+                output_tbl = duckdb.sql(
+                    f"SELECT * FROM output_tbl WHERE {self._where}"
+                ).to_arrow_table()
```

Comment:
It may be necessary to protect from SQL injection attacks:
```suggestion
                output_tbl = duckdb.sql(
                    f"SELECT * FROM output_tbl"
                ).filter(self._where).to_arrow_table()
```